src/libDirectoryService/DirectoryService.cpp

```diff
@@ -1330,8 +1367,104 @@ bool DirectoryService::ProcessInitViewChangeResponse(
     return true;
 }
 
+bool DirectoryService::CleanVariables()
+{
+    LOG_MARKER();
+    m_requesting_last_ds_block = false;
+    {
+        std::lock_guard<mutex> lock(m_MutexCVAllPowConn);
+        m_hasAllPoWconns = true;
+    }
+    m_shards.clear();
+    m_publicKeyToShardIdMap.clear();
+    m_allPoWConns.clear();
+    m_consensusObject.reset();
+    m_consensusBlockHash.clear();
+    {
+        std::lock_guard<mutex> lock(m_mutexPendingDSBlock);
+        m_pendingDSBlock.reset();
+    }
+    {
+        std::lock_guard<mutex> lock(m_mutexAllPOW1);
+        m_allPoW1s.clear();
+    }
+    {
+        std::lock_guard<mutex> lock(m_mutexAllPOW2);
+        m_allPoW2s.clear();
+        m_sortedPoW2s.clear();
+    }
+    {
+        std::lock_guard<mutex> lock(m_mutexMicroBlocks);
+        m_microBlocks.clear();
+    }
+    m_finalBlock.reset();
+    m_finalBlockMessage.clear();
+    m_sharingAssignment.clear();
+    m_viewChangeCounter = 0;
+    // m_initiatedViewChange = false;
+    m_viewChangeEpoch = 0;
+    m_viewChangeRequestTracker.clear();
+    m_viewChangeRequesters.clear();
+    m_mode = IDLE;
+    m_consensusLeaderID = 0;
+    m_consensusID = 0;
+    return true;
+}
+
+void DirectoryService::RejoinAsDS()
+{
+    LOG_MARKER();
+    if (m_mediator.m_lookup->m_syncType == SyncType::NO_SYNC
+        && m_mode == BACKUP_DS)
+    {
+        m_mediator.m_lookup->m_syncType = SyncType::DS_SYNC;
+        m_mediator.m_node->Install(SyncType::DS_SYNC, true);
+        this->StartSynchronization();
+    }
+}
+
+bool DirectoryService::FinishRejoinAsDS()
+{
+    LOG_MARKER();
+    m_mode = BACKUP_DS;
+
+    m_consensusMyID = 0;
+    {
+        std::lock_guard<mutex> lock(m_mediator.m_mutexDSCommitteePubKeys);
+        LOG_GENERAL(INFO,
+                    "m_DSCommitteePubKeys size: "
+                        << m_mediator.m_DSCommitteePubKeys.size());
+        for (auto i = m_mediator.m_DSCommitteePubKeys.begin();
+             i != m_mediator.m_DSCommitteePubKeys.end(); i++)
+        {
+            LOG_GENERAL(INFO, "Loop of m_DSCommitteePubKeys");
+            if (*i == m_mediator.m_selfKey.second)
+            {
+                LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
+                          "My node ID for this PoW1 consensus is "
+                              << m_consensusMyID);
+                break;
+            }
+            m_consensusMyID++;
+        }
+    }
+    // in case the recovery program is under different directory
+    LOG_EPOCHINFO(to_string(m_mediator.m_currentEpochNum).c_str(),
+                  DS_PROMOTE_MSG);
+    RunConsensusOnDSBlock(true);
+    return true;
+}
 #endif // IS_LOOKUP_NODE
 
+bool DirectoryService::ToBlockMessage(unsigned char ins_byte)
+{
+    if (m_mediator.m_lookup->m_syncType != SyncType::NO_SYNC)
+    {
+        return true;
+    }
+    return false;
```

Comment:
return (m_mediator.m_lookup->m_syncType != SyncType::NO_SYNC);
and the function body can be moved to header file

Reply:
The Lookup header doesn't have any function body so far. Better to keep its consistency